clients/drcachesim/scheduler/scheduler.h

```diff
@@ -1066,11 +1076,19 @@ template <typename RecordType, typename ReaderType> class scheduler_tmpl_t {
         bool
         operator()(input_info_t *a, input_info_t *b) const
         {
+            if (a->priority != b->priority)
+                return a->priority < b->priority; // Higher is better.
+            if (!a->order_by_timestamp) {
+                assert(!b->order_by_timestamp);
+                // We use a counter to provide FIFO order.
+                return a->queue_counter > b->queue_counter; // Lower is better.
+            }
             if ((a->reader->get_last_timestamp() - a->base_timestamp) ==
                 (b->reader->get_last_timestamp() - b->base_timestamp)) {
                 // We use a counter to provide FIFO order for same-priority inputs.
-                return a->queue_counter > b->queue_counter;
+                return a->queue_counter > b->queue_counter; // Lower is better.
             }
+            // Lower is better.
             return (a->reader->get_last_timestamp() - a->base_timestamp) >
```

Comment:
It'll be just a tad more clearer if this is moved before the if-block above. To make it clear that timestamp matters more than queue_counter. So something like:

```
            if ((a->reader->get_last_timestamp() - a->base_timestamp) !=
                (b->reader->get_last_timestamp() - b->base_timestamp)) {
                return (a->reader->get_last_timestamp() - a->base_timestamp) > (b->reader->get_last_timestamp() - b->base_timestamp);
            }
            // We use a counter to provide FIFO order for same-priority inputs.
            return a->queue_counter > b->queue_counter; // Lower is better.
```